sormas-api/src/main/java/de/symeda/sormas/api/utils/jurisdiction/SampleJurisdictionHelper.java

```diff
@@ -1,39 +1,45 @@
-/*
- * SORMAS® - Surveillance Outbreak Response Management & Analysis System
- * Copyright © 2016-2020 Helmholtz-Zentrum für Infektionsforschung GmbH (HZI)
- * This program is free software: you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation, either version 3 of the License, or
- * (at your option) any later version.
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
- * GNU General Public License for more details.
- * You should have received a copy of the GNU General Public License
- * along with this program. If not, see <https://www.gnu.org/licenses/>.
- */
-
 package de.symeda.sormas.api.utils.jurisdiction;
 
-import de.symeda.sormas.api.caze.CaseJurisdictionDto;
-import de.symeda.sormas.api.contact.ContactJurisdictionDto;
+import de.symeda.sormas.api.sample.SampleJurisdictionDto;
 import de.symeda.sormas.api.user.JurisdictionLevel;
+import de.symeda.sormas.api.utils.DataHelper;
 
 public class SampleJurisdictionHelper {
 
 	public static boolean isInJurisdiction(
 		JurisdictionLevel jurisdictionLevel,
 		UserJurisdiction userJurisdiction,
-		CaseJurisdictionDto associatedCaseJurisdiction,
-		ContactJurisdictionDto associatedContactJurisdiction) {
-		if (associatedCaseJurisdiction != null) {
-			return CaseJurisdictionHelper.isInJurisdiction(jurisdictionLevel, userJurisdiction, associatedCaseJurisdiction);
+		SampleJurisdictionDto sampleJurisdiction) {
+
+		if (sampleJurisdiction.getReportingUserUuid() != null
+			&& DataHelper.equal(userJurisdiction.getUuid(), sampleJurisdiction.getReportingUserUuid())) {
+			return true;
+		}
+
+		switch (jurisdictionLevel) {
+		case NATION:
+			return true;
+		case NONE:
+		case COMMUNITY:
+		case DISTRICT:
+		case REGION:
+		case HEALTH_FACILITY:
+		case POINT_OF_ENTRY:
+			return false;
+		case LABORATORY:
+		case EXTERNAL_LABORATORY:
+			return sampleJurisdiction.getLabUuid() != null
+				&& DataHelper.equal(sampleJurisdiction.getLabUuid(), userJurisdiction.getHealthFacilityUuid());
+		}
+
+		if (sampleJurisdiction.getCaseJurisdiction() != null) {
```

Comment:
If I'm not mistaken, this code is never reached because the switch above will always return, which would very likely lead to bugs (because users should be allowed to edit samples of cases that they can edit no matter their jurisdiction level as long as they have the SAMPLE_EDIT user right)